google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -174,6 +174,92 @@ void WriteObject(google::cloud::storage::Client client, int& argc,
   (std::move(client), bucket_name, object_name, desired_line_count);
 }
 
+void GenerateEncryptionKey(google::cloud::storage::Client client, int& argc,
+                           char* argv[]) {
+  if (argc != 1) {
+    throw Usage{"generate-encryption-key"};
+  }
+  //! [generate encryption key] [START generate_encryption_key_base64]
+  // Create a pseudo-random number generator (PRNG), this is included for
+  // demonstration purposes only. You should consult your security team about
+  // best practices to initialize PRNG. In particular, you should verify that
+  // the C++ library and operating system provide enough entropy to meet the
+  // security policies in your organization.
+
+  // Use the Mersenne-Twister Engine in this example:
+  //   https://en.cppreference.com/w/cpp/numeric/random/mersenne_twister_engine
+  // Any C++ PRNG can be used below, the choice is arbitrary.
+  using GeneratorType = std::mt19937_64;
+
+  // Create the default random device to fetch entropy.
+  std::random_device rd;
+
+  // Compute how much entropy we need to initialize the GeneratorType:
+  constexpr auto kRequiredEntropyWords =
+      GeneratorType::state_size *
+      (GeneratorType::word_size / std::numeric_limits<unsigned int>::digits);
+
+  // Capture the entropy bits into a vector.
+  std::vector<unsigned long> entropy(kRequiredEntropyWords);
+  std::generate(entropy.begin(), entropy.end(), [&rd] { return rd(); });
+
+  // Create the PRNG with the fetched entropy.
+  std::seed_seq seed(entropy.begin(), entropy.end());
+  GeneratorType gen(seed);
+
+  namespace gcs = google::cloud::storage;
+  gcs::EncryptionKeyData data = gcs::CreateKeyFromGenerator(gen);
+
+  std::cout << "Base64 encoded key = " << data.key << "\n"
+            << "Base64 encoded SHA256 of key = " << data.sha256 << std::endl;
+  //! [generate encryption key] [END generate_encryption_key_base64]
+}
+
+void WriteEncryptedObject(google::cloud::storage::Client client, int& argc,
+                          char* argv[]) {
+  if (argc != 4) {
+    throw Usage{
+        "write-encrypted-object <bucket-name> <object-name> <raw-aes256-key>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  auto object_name = ConsumeArg(argc, argv);
+  auto raw_aes256_key = ConsumeArg(argc, argv);
```

Comment:
IIUC, I think it would be easier to use the base64 format instead of raw format given the generator provides a base64 encoded key. _Similar feedback from DownloadEncryptedObject_

Reply:
Remember that it won't be easy for users to generate a Base64 encoded key in C++, no standard library to do that.  Maybe we should accept Base64 encoded keys *in addition* to accepting raw keys?

Regardless, would you prefer the examples to take a base64 key?  The advantage of raw strings is that they require less typing if the user wants to run the test themselves:

```sh
./storage_object_samples write-encrypted-object my-bucket my-object my-secret-key...
```

Whereas if we require them to give us base64 keys then they have to use:

```sh
KEY="echo my-secret-key-padded-to-32-chars | openssl base64" 
./storage_object_samples write-encrypted-object my-bucket my-object $KEY
```

Reply:
w.r.t accepting base64 encoded keys, I think that sounds reasonable given the library has a dependency on library with base64 support. so `gcs::EncryptionKey::FromBase64". **I'm thinking not all keys will be representable in text**.

w.r.t to how the key is passed in with the CLI. I look at the examples as getting started with the library. I didn't expect a developer to depend on this CLI. 

The User Journey I expected was:
```
./storage_object_samples generate-encryption-key
# Grab the base64 key
./storage_object_samples write-encrypted-object my-bucket my-object <<paste base64 encoded key>>
```
The issue I'm understanding is the sample is focused on using raw keys versus base64 encoded keys. While I'm looking at how the CLI would be used. I think it makes sense to show only raw aes256 keys given the C++ context, but I'd question how these keys are eventually stored in a database. Possibly as a base64 encoded string?

Reply:
Done.